src/src/Helpers/Networking.cpp

```diff
@@ -1575,24 +1575,33 @@ int http_authenticate(const String& logIdentifier,
       //                                   channel number ┘    |        └ received values
       //                                                   field number (only available for a "single-value-event")
       // In rules you can grep the reply by "On ThingspeakReply Do ..."
-     
-      if (httpCode == 200 && equals(host, F("api.thingspeak.com")) && uri.endsWith(F("/last.csv"))) {
-        String result = http.getString(); 
-        const int posTimestamp = result.lastIndexOf(':');
-        if (posTimestamp >= 0) { 
-          result = parseStringToEndKeepCase(result.substring(posTimestamp), 3);
-          if (uri.indexOf(F("fields")) >= 0) { //when there is a single field call add the field number before the value
-            result = parseStringKeepCase(uri, 4, '/') + "," + result;
-          }
-          eventQueue.addMove(strformat(
-            F("ThingspeakReply=%s,%s"),                                            
+      //-----------------------------------------------------------------------------------------------------------------------------
+      // 2024-02-05 - Added the option to get a single value of a field or all values of a channel at a certain time (not only the last entry)
+      // Examples:
+      // Single channel: "sendtohttp,api.thingspeak.com,80,channels/1637928/fields/1.csv?end=2024-01-01%2023:59:00&results=1"
+      // => gets the value of field 1 at (or the last entry before) 23:59:00 of the channel 1637928
+      // All channels: "sendtohttp,api.thingspeak.com,80,channels/1637928/feeds.csv?end=2024-01-01%2023:59:00&results=1"
+      // => gets the value of each field of the channel 1637928 at (or the last entry before) 23:59:00 
+      //-----------------------------------------------------------------------------------------------------------------------------
+
+    if (httpCode == 200 && equals(host, F("api.thingspeak.com")) && (uri.endsWith(F("/last.csv")) || (uri.indexOf(F("results=1")) >= 0 && uri.indexOf(F(".csv")) >= 0))){
```

Comment:
Check seems to be wrong.
You need to have the host to match and then the url can be 2 options.

So maybe have an extra layer of if's where you first check for httpcode and host and then the if with the uri checks.

Reply:
actually it works like this...

here simplified to see the important brackets:
`(http code && host && (option1 || (option2.1  && option2.2)) `

Reply:
Ah yep now I see it....